### PR TITLE
[FP16] Add a feature flag for FP16.

### PR DESCRIPTION
**Pull request by brendandahl**
Ensure the "fp16" feature is enabled for FP16 instructions.

### CODE REVIEW

File: src/wasm/wasm.cpp [hunk 1]
```diff
@@ -56,6 +56,7 @@ const char* StringsFeature = "strings";
 const char* MultiMemoryFeature = "multimemory";
 const char* TypedContinuationsFeature = "typed-continuations";
 const char* SharedEverythingFeature = "shared-everything";
+const char* FP16Feature = "fp16";
```
Comment (tlively) [inline src/wasm/wasm.cpp:59]: Just to double check, does this match the feature name used by LLVM in the custom features section?

**Review by tlively — COMMENTED** [review]
We should also update fuzzing.cpp to gate generation of the fp16 ops behind the feature flag.

File: src/wasm/wasm.cpp (hunk 1 above)
Comment (brendandahl) [reply src/wasm/wasm.cpp:59]: It will after https://github.com/llvm/llvm-project/pull/105434

**Review by kripken — APPROVED** [review]

**Comment by brendandahl** [comment]
> We should also update fuzzing.cpp to gate generation of the fp16 ops behind the feature flag.

Preference for in this PR or follow up?

**Comment by tlively** [comment]
Ideally in this one. Thanks!

**Comment by brendandahl** [comment]
Added some fuzzing. FWW, I noticed we're missing relaxed simd support and splat instructions for floats.

**Review by kripken — APPROVED** [review]
lgtm with fuzzing.

Side note, for incremental updates to a PR I would ask that we all try to use additional commits, rather than force-pushing a single commit. Incremental commits make incremental review easier, at least for me.

**Comment by brendandahl** [comment]
I try not to force push if I've rebased in upstream, but for small commits it seems the github "compare" link on the force pushes is as good as a separate commit.

**Comment by brendandahl** [comment]
I guess the review UI is not on the compare link. I'll try to stick with separate commits.

**Comment by kripken** [comment]
I wasn't aware of the "compare" button, thanks! Looks useful.

But, yeah, it does seem to be missing the review UI, unfortunately, which commits do have.

**Pull request merged**